zendesk/automation.go

```diff
@@ -0,0 +1,172 @@
+package zendesk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// AutomationCondition zendesk automation condition
+//
+// ref: https://developer.zendesk.com/rest_api/docs/core/automations#conditions-reference
+type AutomationCondition struct {
```

Comment:
So obviously alot of this code is very similar to what is in trigger, and personally i think it all looks fine.

that said [dupl](https://github.com/mibk/dupl) doesn't seem to agree with me, and i don't know what would be the best way to make it happy. To me it looks like our options are:

1. disable dupl (either for just this file or the entire project)
2. Refactor trigger interface to make the structs shared between the two use more generic names 
3. refactor this file to use all the structs from trigger.

Personally i think i lean toward 1 because i don't really want to go and mess with the trigger API being that we don't really know who all is using it at this point. 

Anyone else have anythoughts @nukosuke @RWejlgaard

Reply:
yeah, I agree. 

Honestly, I expect to simplify only syntax review process from golangci. So, I think it fine to disable `dupl` for the entire project.

At this point, we might want to take "2. Ignore it and merge anyway".

@RWejlgaard I'll merge this pr if you would like to.

Reply:
Yes that sounds fine to me as well

Reply:
👍